repo-agent/review-ui/review-api/main.go

```diff
@@ -87,356 +107,607 @@ func (w bodyLogWriter) Write(b []byte) (int, error) {
+	}
+}
+
+func initOAuth() {
+	clientID := os.Getenv("GITHUB_CLIENT_ID")
+	clientSecret := os.Getenv("GITHUB_CLIENT_SECRET")
+
+	if clientID == "" || clientSecret == "" {
+		log.Println("Warning: GITHUB_CLIENT_ID or GITHUB_CLIENT_SECRET not set. OAuth will not work.")
+	}
 
-		c.Next() // Process the request and generate the response
+	oauthConf = &oauth2.Config{
+		ClientID:     clientID,
+		ClientSecret: clientSecret,
+		Scopes:       []string{"read:user", "user:email"},
+		Endpoint:     githuboauth.Endpoint,
+	}
 
-		log.Printf("Response Status: %d\n", c.Writer.Status())
-		log.Printf("Response Headers: %v\n", c.Writer.Header())
-		log.Printf("Response Body: %s\n", blw.body.String())
+	b := make([]byte, 16)
+	if _, err := rand.Read(b); err != nil {
+		log.Fatalf("Failed to generate random OAuth state: %v", err)
 	}
+	oauthState = base64.URLEncoding.EncodeToString(b)
 }
 
 func main() {
-	// Redis client
 	redisAddr := os.Getenv("REDIS_ADDR")
 	if redisAddr == "" {
 		redisAddr = "localhost:6379"
 	}
-	rdb = redis.NewClient(&redis.Options{
-		Addr: redisAddr,
-	})
+	rdb = redis.NewClient(&redis.Options{Addr: redisAddr})
 
-	// Kubernetes client
 	config, err := rest.InClusterConfig()
 	if err != nil {
-		log.Printf("Failed to get in-cluster config, trying local config: %v", err)
 		kubeconfig := os.Getenv("KUBECONFIG")
 		if kubeconfig == "" {
 			kubeconfig = os.Getenv("HOME") + "/.kube/config"
 		}
 		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
 		if err != nil {
-			log.Fatalf("Failed to get local kubeconfig: %v", err)
+			log.Fatalf("Failed to get kubeconfig: %v", err)
 		}
 	}
 	k8sClient, err = dynamic.NewForConfig(config)
 	if err != nil {
-		log.Fatalf("Failed to create kubernetes client: %v", err)
+		log.Fatalf("Failed to create dynamic client: %v", err)
 	}
-
-	// Ping redis to ensure connection
-	_, err = rdb.Ping(context.Background()).Result()
+	k8sClientset, err = kubernetes.NewForConfig(config)
 	if err != nil {
-		log.Fatalf("Failed to connect to Redis: %v", err)
+		log.Fatalf("Failed to create clientset: %v", err)
 	}
 
-	// Pre-populate mock data in Redis
-	populateMockData()
+	initOAuth()
 
-	// Gin router
-	router := gin.Default()
+	if _, err := rdb.Ping(context.Background()).Result(); err != nil {
+		log.Fatalf("Failed to connect to Redis: %v", err)
+	}
 
-	// Add middleware to log requests and responses
+	router := gin.Default()
+	store := cookie.NewStore([]byte("secret"))
+	router.Use(sessions.Sessions(sessionName, store))
 	router.Use(RequestLoggerMiddleware())
 	router.Use(ResponseLoggerMiddleware())
 
-	// API routes
+	router.GET("/api/auth/login", authLogin)
+	router.GET("/api/auth/callback", authCallback)
+	router.GET("/api/auth/status", authStatus)
+	router.POST("/api/auth/logout", authLogout)
+
 	api := router.Group("/api")
+	api.Use(authMiddleware())
 	{
 		api.GET("/repos", getRepos)
-		api.GET("/repo/:namespace/:repo/prs", getPRs)
-		api.POST("/repo/:namespace/:repo/prs/:id/draft", saveDraft)
```

Comment:
curious. If we remove namespace from the URL does this make the api-server single tenant ?